src/data/reports/glass-admin/GLASSDataMaintenanceDefaultRepository.ts

```diff
@@ -0,0 +1,195 @@
+
+    async getUserModules(config: Config): Promise<GLASSModule[]> {
+        const modules = await this.getModules();
+
+        const userGroups = config.currentUser.userGroups;
+        const userGroupIds = userGroups.map(userGroup => userGroup.id);
+
+        const userModules = modules.filter(module => {
+            const moduleUserGroupIds = module.userGroups.approveAccess?.map(userGroup => userGroup.id) ?? [];
+
+            return _.some(moduleUserGroupIds, moduleUserGroupId => userGroupIds.includes(moduleUserGroupId));
+        });
+
+        return userModules;
+    }
+
+    async delete(namespace: string, items: Id[]): Promise<void> {
+        await this.deleteFileResources(items);
+
+        const uploads = await this.getUploads(namespace);
+        const updatedUploads = this.updateStatusById(items, uploads, "DELETED");
+        return await this.globalStorageClient.saveObject<GLASSDataMaintenanceItem[]>(namespace, updatedUploads);
+    }
+
+    async getColumns(namespace: string): Promise<string[]> {
+        const columns = await this.storageClient.getObject<string[]>(namespace);
+
+        return columns ?? [];
+    }
+
+    async saveColumns(namespace: string, columns: string[]): Promise<void> {
+        return this.storageClient.saveObject<string[]>(namespace, columns);
+    }
+
+    private async getCountries(): Promise<NamedRef[]> {
+        const { organisationUnits } = await this.api
+            .get<{ organisationUnits: NamedRef[] }>("/organisationUnits", {
+                fields: "id, name",
+                filter: "level:eq:3",
+                paging: false,
+            })
+            .getData();
+
+        return organisationUnits;
+    }
+
+    private async getModules(): Promise<GLASSModule[]> {
+        const modules =
+            (await this.globalStorageClient.getObject<GLASSModule[]>(Namespaces.DATA_SUBMISSSIONS_MODULES)) ?? [];
+
+        return modules
+            .map(module => ({
+                id: module.id,
+                name: module.name,
+                userGroups: module.userGroups,
+            }))
+            .filter(module => module.name !== earModule);
+    }
+
+    private async getUploads(namespace: string): Promise<GLASSDataMaintenanceItem[]> {
+        const uploads = (await this.globalStorageClient.getObject<GLASSDataMaintenanceItem[]>(namespace)) ?? [];
+
+        return uploads;
+    }
+
+    private async deleteFileResources(items: Id[]): Promise<void> {
+        _.forEach(items, async item => {
+            try {
+                await this.api.delete(`/documents/${item}`).getData();
+            } catch (error) {
+                console.debug("File does not exist");
+            }
+        });
+    }
+
+    private getFilteredFiles(files: GLASSDataMaintenanceItem[], orgUnits: NamedRef[], module: Module) {
+        const lastDataSubmissionYear = new Date().getFullYear() - 1; // last submission year is the previous year
+
+        return _(files)
+            .map(upload => ({
```

Comment:
be consistent: files/upload, should be the same, one plural, the other singular.